### PR TITLE
64194 betamocks map sus

### PR DESCRIPTION
**Pull request by achievedaniel**
## Summary

- This PR adds the MAP STS endpoint details to the betamocks configuration.
- This facilitates for rapid development for ToU and other features interacting with MAP STS API; otherwise, development requires use of SOCKS proxy.
- This work is being completed by @vsp-identity

## Related issue

- department-of-veterans-affairs/va.gov-team#64191

## Related PR

- https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/420

## Testing done

Prior to this change, any call to the STS server would fail. This PR enables betamocks for these calls, so they should succeed and return a mocked response.

### Test Steps

- Pull the latest changes to the `vets-api-mockdata` repo.
  - Changes should include additions from PR https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/420
  - If this PR isn't merged yet, check out the branch locally.
- Open a rails console.
- Require the SUS service.
```ruby
require 'mobile_application_platform/sign_up/service'
```
- Call the service different methods with the required parameters:
```ruby
MobileApplicationPlatform::SignUp::Service.new.status(icn:)
MobileApplicationPlatform::SignUp::Service.new.agreements_accept(icn:, signature_name:, version:)
MobileApplicationPlatform::SignUp::Service.new.agreements_decline(icn:)
MobileApplicationPlatform::SignUp::Service.new.update_provisioning(icn:, first_name:, last_name:, mpi_gcids:)
```
- Verify the response is mocked.

## Screenshots

Not relevant.

## What areas of the site does it impact?

This does not have an impact to production facilities of the site; it only affects local development.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Please advise on best practices using the betamocks library and whether additional configuration is necessary.

### CODE REVIEW

**Review by bosawt — CHANGES_REQUESTED** [review]
I'm not sure where my previous review went? Can you update these to use `cache_multiple_responses` with a regex that mocks different ICNs differently?

**Comment by achievedaniel** [comment]
> I'm not sure where my previous review went? Can you update these to use `cache_multiple_responses` with a regex that mocks different ICNs differently?

From what I am seeing in the Betamocks documentation the `cache_multiple_responses` is for recording responses only. Meaning that for every ICN we will need to record a new response which will add a file to the service mock directory. 

I am not seeing a feature where you can dynamically change the response based on the params passed. We will need to add that to Betamocks if we want to return custom dynamic responses.

File: config/settings.yml [hunk 1]
```diff
@@ -34,6 +34,10 @@ mobile_application_platform:
   client_key_path: spec/fixtures/mobile_application_platform/oauth.key
   sign_up_service_url: https://cerner.apps-staging.va.gov
   sign_up_service_provisioning_api_key: abcd1234
+  sts:
```
Comment (dickdavis) [inline config/settings.yml:37]: You'll need to rebase to pull in my commit and then drop your change (`sts: mock: true`) when working through the merge conflict.

**Comment by bosawt** [comment]
@achievedaniel Can you remove all of the extra groups in the reviewers list that were added whenever what must have been an improper merge occurred? I don't want 30 groups to get pinged every time we add a change or comment in this PR

File: config/settings.yml [hunk 2]
```diff
@@ -350,7 +352,12 @@ modules_appeals_api:
     daily_error:
       enabled: false
     weekly_error:
-      enabled: false
+      # See features.yml for enablement
```
Comment (bosawt) [inline config/settings.yml:355]: There are extra changes in this file that shouldn't be part of this PR

File: config/betamocks/services_config.yml [hunk 3]
```diff
@@ -934,3 +934,32 @@
...
+  # Sign Up Service API
+- :name: "MAP SUS"
+  :base_uri: <%= "#{URI(Settings.mobile_application_platform.sign_up_service_url).host}:#{URI(Settings.mobile_application_platform.sign_up_service_url).port}" %>
+  :endpoints:
+    - :method: :get
+      :path: "/signup/v1/patients/*/status/summary"
+      :file_path: "map/sign_up_service/status"
+      :cache_multiple_responses:
+        :uid_location: url
+        :uid_locator: '\/signup\/v1\/patients\/(\d{10}[V]\d{6})\/status\/summary'
```
Comment (bosawt) [inline config/betamocks/services_config.yml:947]: I would make the capture regex for the icn just anything in between `patients` and `status`.  Yours might be more accurate but it will unexpectedly attempt to mock with a file called `.yml` if the given ICN doesn't exactly match 10 digits, 'V', 6 digits

Comment (bosawt) [reply config/betamocks/services_config.yml:947]: (And do something similar for the rest of these)

**Review by bosawt — CHANGES_REQUESTED** [review]
Some changes requested as described in my comments

**Comment by bosawt** [comment]
This work is completed here: https://github.com/department-of-veterans-affairs/vets-api/pull/14245
